applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -256,280 +228,352 @@ def RunHROM(self, mu_run=[None], use_full_model_part = False):
             parameters = KratosMultiphysics.Parameters(parameter_file.read())
-
-        SnapshotsMatrix = []
+        BasisOutputProcess = None
         for Id, mu in enumerate(mu_train):
-            parameters_copy = self.UpdateProjectParameters(parameters.Clone(), mu)
-            parameters_copy = self._AddBasisCreationToProjectParameters(parameters_copy)
-            parameters_copy = self._StoreResultsByName(parameters_copy,'HROM_Fit',mu,Id)
-            materials_file_name = parameters_copy["solver_settings"]["material_import_settings"]["materials_filename"].GetString()
-            self.UpdateMaterialParametersFile(materials_file_name, mu)
-            model = KratosMultiphysics.Model()
-            analysis_stage_class = type(SetUpSimulationInstance(model, parameters_copy))
-            simulation = self.CustomizeSimulation(analysis_stage_class,model,parameters_copy)
-            simulation.Run()
-            self.QoI_Fit_HROM.append(simulation.GetFinalData())
-            for process in simulation._GetListOfOutputProcesses():
-                if isinstance(process, CalculateRomBasisOutputProcess):
-                    BasisOutputProcess = process
-            SnapshotsMatrix.append(BasisOutputProcess._GetSnapshotsMatrix()) #TODO add a CustomMethod() as a standard method in the Analysis Stage to retrive some solution
-        SnapshotsMatrix = np.block(SnapshotsMatrix)
+            in_database, _ = self.data_base.check_if_in_database("HROM_Fit", mu)
+            if not in_database:
+                parameters_copy = self.UpdateProjectParameters(parameters.Clone(), mu)
+                parameters_copy = self._AddBasisCreationToProjectParameters(parameters_copy)
+                parameters_copy = self._StoreResultsByName(parameters_copy,'HROM_Fit',mu,Id)
+                materials_file_name = parameters_copy["solver_settings"]["material_import_settings"]["materials_filename"].GetString()
+                self.UpdateMaterialParametersFile(materials_file_name, mu)
+                model = KratosMultiphysics.Model()
+                analysis_stage_class = type(SetUpSimulationInstance(model, parameters_copy))
+                simulation = self.CustomizeSimulation(analysis_stage_class,model,parameters_copy)
+                simulation.Run()
+                self.QoI_Fit_HROM.append(simulation.GetFinalData())
+                for process in simulation._GetListOfOutputProcesses():
+                    if isinstance(process, CalculateRomBasisOutputProcess):
+                        BasisOutputProcess = process
+                SnapshotsMatrix = BasisOutputProcess._GetSnapshotsMatrix() #TODO add a CustomMethod() as a standard method in the Analysis Stage to retrive some solution
+                self.data_base.add_to_database("HROM_Fit", mu, SnapshotsMatrix)
+
+        self.GenerateDatabaseSummary()
+
 
-        return SnapshotsMatrix
 
 
-    def __LaunchTestFOM(self, mu_test):
+    def _LaunchTestFOM(self, mu_test):
```

Comment:
Wouldn't make sense to unify this function and _LaunchTrainFOM? As they both just solve the FOM for the given parameter. Also, couldn't all FOM snapshots go into a unique SQLite table?, instead of having one for test and one for train

Reply:
Methods available are now _LaunchFOM(), LaunchROM() and LaunchHROM()